### PR TITLE
2625 port chainmanager from poc

### LINKED ISSUE #2625: Port ChainManager from PoC
## Description
We created a ChainManager class in the quiet sandbox proof of concept, and we should port that demo work into a more finalized version for integration. Remove anything unnecessary for integration from the chain manager, and dd the LFA wrapper as a chain manager class in the backend.

## Acceptance Criteria
1. ChainManager class created
1. Includes methods for creating Sigchains, setting active chains, and initialization

### PR DESCRIPTION
**Pull request by adrastaea**
Adds basic Sigchain classes and tests

### Pull Request Checklist

- [x] I have linked this PR to a related GitHub issue.
- [x] I have added a description of the change (and Github issue number, if any) to the root [CHANGELOG.md](https://github.com/TryQuiet/quiet/blob/develop/CHANGELOG.md).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

### CODE REVIEW

File: packages/backend/src/nest/auth/services/crypto/crypto.service.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
+/**
```
Comment (islathehut) [inline packages/backend/src/nest/auth/services/crypto/crypto.service.ts:1]: This looks like an older version, you'll want the one from the `isla-perf-test` branch

https://github.com/TryQuiet/auth/blob/f7d1b374d4e6a1ef01160c8de6e6ab06773a49fb/demos/quiet-sandbox/src/auth/services/crypto/cryptoService.ts

File: packages/backend/src/nest/auth/services/dm/dm.service.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (islathehut) [inline packages/backend/src/nest/auth/services/dm/dm.service.ts]: I would honestly delete this, my attempts at creating DMs were mostly experimental.  I'm not sure if this really maps to how we'll do DMs in the future.

File: packages/backend/src/nest/auth/services/invites/invite.service.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  public getById(id: Base58): InvitationState {
+    return this.sigChain.team.getInvitation(id)
+  }
+
+  public static generateProof(seed: string): ProofOfInvitation {
+    return SigChain.lfa.invitation.generateProof(seed)
+  }
+
+  public validateProof(proof: ProofOfInvitation): boolean {
```
Comment (islathehut) [inline packages/backend/src/nest/auth/services/invites/invite.service.ts:65]: You can delete `validateProof`, `admitUser`, and `admitMemberFromInvite`.  This is handled by the LFA `Connection` class we use in libp2p.

**Review by islathehut — CHANGES_REQUESTED** [review]
Just some minor things.  The main one is updating the crypto service to the latest on the `isla-perf-test` branch as that uses the correct symmetric encryption.

File: packages/backend/src/nest/auth/services/invites/invite.service.ts (hunk 3 above)
Comment (adrastaea) [reply packages/backend/src/nest/auth/services/invites/invite.service.ts:65]: These are useful in testing so I'm going to leave them in for now I think

**Review by islathehut — APPROVED** [review]

**Pull request merged**